geode-redis/src/main/java/org/apache/geode/redis/internal/netty/ExecutionHandlerContext.java

```diff
@@ -113,17 +114,19 @@ public ChannelFuture writeToChannel(RedisResponse response) {
   @Override
   public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
     Command command = (Command) msg;
-    try {
-      if (logger.isDebugEnabled()) {
-        logger.debug("Executing Redis command: {}", command);
+    command.setChannelHandlerContext(ctx);
```

Comment:
I think this might read a bit better like this:
```
    synchronized (commandQueue) {
      boolean isEmpty = commandQueue.isEmpty();
      boolean isAsync = command.getCommandType().isAsync();
      if (!isEmpty) {
        commandQueue.offer(command);
        return;
      }
      if (!isAsync) {
        executeCommand(command);
      } else {
        commandQueue.offer(command);
        startAsyncCommandExecution(command);
      }
    }
```

Reply:
Yes, and since isEmpty and isAsync are only used once I ended up changing it to this:
`    synchronized (commandQueue) {
      if (!commandQueue.isEmpty()) {
        commandQueue.offer(command);
        return;
      }
      if (command.getCommandType().isAsync()) {
        commandQueue.offer(command);
        startAsyncCommandExecution(command);
        return;
      }
      executeCommand(command);
    }
`